torch/_inductor/lowering.py

```diff
@@ -1736,9 +1739,35 @@ def register_fallback(op_overload):
         )
 
     if isinstance(op, torch._ops.OpOverloadPacket):
-        for ol in op.overloads():
+        skip_overloads = []
+        overloads = []
+        for ol in op.overloads_with_kernels():
             op_overload = getattr(op, ol)
-            register_fallback(op_overload)
+            if op_overload in decompositions:
+                skip_overloads.append(op_overload)
+            else:
+                overloads.append(op_overload)
+        if overloads:
+            # Having some overloads in decompositions and others not,
+            # is very untypical. So, we'll be maximally verbose here.
+            for op_overload in skip_overloads:
+                log.warning(
+                    "Skipping to register fallback for %s that is in decompositions.",
+                    op_overload,
+                )
+            for op_overload in overloads:
+                if skip_overloads:
+                    log.warning(
+                        "Registering fallback for %s while there exists other overloads in decompositions.",
+                        op_overload,
+                    )
+                register_fallback(op_overload)
+        else:
+            log.warning(
```

Comment:
Yes, that makes sense. However, we first have to resolve the following warnings (the output is obtained with the angle-lowering PR, otherwise there are `angle` related warnings as well):
```
[2024-01-17 16:59:53,357] torch._inductor.lowering: [WARNING] Skipping to register fallback for aten.randint.default that is in decompositions.
[2024-01-17 16:59:53,357] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.generator while there exists other overloads in decompositions.
[2024-01-17 16:59:53,357] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.low while there exists other overloads in decompositions.
[2024-01-17 16:59:53,357] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.low_generator while there exists other overloads in decompositions.
[2024-01-17 16:59:53,357] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.out while there exists other overloads in decompositions.
[2024-01-17 16:59:53,358] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.generator_out while there exists other overloads in decompositions.
[2024-01-17 16:59:53,358] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.low_out while there exists other overloads in decompositions.
[2024-01-17 16:59:53,358] torch._inductor.lowering: [WARNING] Registering fallback for aten.randint.low_generator_out while there exists other overloads in decompositions.
[2024-01-17 16:59:53,358] torch._inductor.lowering: [WARNING] make_fallback(aten.convolution_backward): all operation overloads are in decompositions. Delete the corresponding `make_fallback` line.
[2024-01-17 16:59:53,416] torch._inductor.lowering: [WARNING] make_fallback(aten.soft_margin_loss_backward): all operation overloads are in decompositions. Delete the corresponding `make_fallback` line.
[2024-01-17 16:59:53,417] torch._inductor.lowering: [WARNING] make_fallback(aten.masked_scatter): all operation overloads are in decompositions. Delete the corresponding `make_fallback` line.
```

So, I suggest making these warnings as errors a bit later when the warnings are resolved.

Reply:
Do you plan to fix these in this stack? Otherwise I don't think that spamming warnings is the best idea.

Reply:
yes